frontend/components/Home/Dashboard/Modals/ModalLeaveStudentCourse.tsx

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import { Modal, Button } from "semantic-ui-react";
+import { leaveCourse } from "../../../../hooks/data-fetching/dashboard";
+import { mutateFunction, UserMembership } from "../../../../types";
+
+interface ModalLeaveStudentCourseProps {
+    open: boolean;
+    leaveMembership?: UserMembership;
+    closeFunc: () => void;
+    mutate: mutateFunction<UserMembership[]>;
+}
+
+const ModalLeaveStudentCourse = (props: ModalLeaveStudentCourseProps) => {
+    const { open, leaveMembership, closeFunc, mutate } = props;
+
+    const leaveFunc = async () => {
+        await leaveCourse(

```

Comment:
leaveCourse could throw an error here - consider handling it like ModalAddInstructorCourse.tsx